pilot/pkg/security/authz/policy/helper.go

```diff
@@ -216,8 +238,11 @@ func Verify(got *envoy_rbac.RBAC, want map[string][]string) error {
 			if !ok {
 				err = multierror.Append(err, fmt.Errorf("not found rule %q", key))
 			} else {
-				// FIXME(pitlv2109/yangmingzhu): This doesn't always work because |actualStr| might contain
-				// more stuff than |values| from |want|. Need to check both ways.
+				// Enable principal count check if principals are important for the current test (e.g. generating RBAC config based
+				// on trust domain aliases).
+				if arePrincipalsImportant {
+					err = checkPrincipalCounts(key, actualStr, values)
```

Comment:
Did you check the expected principals already? Why do you need to check principal count?

Reply:
This comment explains this in more detail https://github.com/istio/istio/pull/18055#discussion_r337757306

Reply:
@liminw I added a commit to change the name to `needToCheckPrincipals` instead, also another commit to reflect the presence match PR that got merged recently. PTAL and let me know if you have any more comments. Thanks!